pkg/debug/start_debug.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package debug
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s "k8s.io/client-go/kubernetes"
+)
+
+var getPodFromDeployment = `
+kubectl -n %s get pod | grep %s | awk '{ print $1 }'
+`
+
+var scriptForPodDeletion = `
+kubectl -n %s  wait --for=delete pod/%s --timeout=60s
+`
+
+func StartDebug(operatorNamespace, clusterNamespace, deploymentName, flagValue string) {
+	// Get Kubernetes Client
+	_, clientset, _ := k8sutil.GetKubeClient()
+
+	deployment, deploymentExist := verifyDeploymentExists(clientset, clusterNamespace, deploymentName)
+	if !deploymentExist {
+		os.Exit(1)
+	}
+
+	deploymentSpec := deployment.Spec
+
+	if flagValue != "" {
+		deploymentSpec.Template.Spec.Containers[0].Image = flagValue
+	}
+
+	labels := deployment.Labels
+	labels["ceph.rook.io/do-not-reconcile"] = "true"
+
+	deploymentSpec.Template.Spec.Containers[0].LivenessProbe = nil
+	deploymentSpec.Template.Spec.Containers[0].StartupProbe = nil
+
+	if flagValue != "" {
+		log.Printf("setting debug image to %s\n", flagValue)
+		deploymentSpec.Template.Spec.Containers[0].Image = flagValue
+	}
+
+	fmt.Println("setting debug command to main container")
+
+	deploymentSpec.Template.Spec.Containers[0].Command = []string{"sleep", "infinity"}
+	deploymentSpec.Template.Spec.Containers[0].Args = []string{}
+
+	if err := updateDeployment(clientset, clusterNamespace, deployment); err != nil {
+		os.Exit(1)
+	}
+
+	deployment_Pod_Name, err := podRunningWithLabel(clientset, clusterNamespace, deploymentSpec)
+	if deployment_Pod_Name == "" || err != nil {
+		os.Exit(1)
+	}
+
+	if err := getAndUpdatetheScale(clientset, clusterNamespace, deployment.Name, 0); err != nil {
+		os.Exit(1)
+	}
+
+	fmt.Printf("waiting for the deployment pod %s to be deleted\n", deployment_Pod_Name)
+
+	waitForPodToDeleted := exec.Command("/bin/bash",
+		"-e", // Exit immediately if a command exits with a non-zero status.
+		"-m", // Terminal job control, allows job to be terminated by SIGTERM
+		"-c", // Command to run
+		fmt.Sprintf(scriptForPodDeletion, clusterNamespace, deployment_Pod_Name),
+	)
+	_, err = waitForPodToDeleted.Output()
```

Comment:
Instead of this script, can we use a similar K8s helper to wait for the pod to be deleted?